src/annotator/integrations/test/html-metadata-test.js

```diff
@@ -405,5 +405,24 @@ describe('HTMLMetadata', () => {
 
       assert.equal(doc.uri(), canonicalLink.href);
     });
+
+
+    it('should log an error if URI is not decodable', () => {
+      // Arrange
+      // '%%CUST_ID%%' is an invalid escape sequence, so it will throw a URIError when decoded
+      const badURI = 'https://example.com/?foo=%%CUST_ID%%' 
+      sinon.stub(testDocument, '_getDocumentHref').returns(badURI); 
+      const consoleErrorSpy = sinon.spy(console, 'error');
+
+      // Act
+      testDocument.uri(); // The method that triggers the try...catch
+      
+      // Assert
+      assert(consoleErrorSpy.calledOnce);
+      assert(consoleErrorSpy.calledWith('Error decoding URI:', sinon.match.instanceOf(URIError)));
+    
+      // Cleanup
+      console.error.restore();
```

Comment:
I think we can skip the "Arrange", "Act", "Assert" and "Cleanup" comments, as they fall under the "what" and not the "why", and hence, they add little value.

```suggestion
      // '%%CUST_ID%%' is an invalid escape sequence, so it will throw a URIError when decoded
      const badURI = 'https://example.com/?foo=%%CUST_ID%%' 
      sinon.stub(testDocument, '_getDocumentHref').returns(badURI); 
      const consoleErrorSpy = sinon.spy(console, 'error');

      testDocument.uri(); // The method that triggers the try...catch
      
      assert(consoleErrorSpy.calledOnce);
      assert(consoleErrorSpy.calledWith('Error decoding URI:', sinon.match.instanceOf(URIError)));
    
      console.error.restore();
```